### PR TITLE
Dns Runtime Config

### PR DESCRIPTION
**Pull request by nagiesek**
This adds the dns capabilities option

### CODE REVIEW

File: pkg/server/sandbox_stop.go [hunk 1]
```diff
@@ -123,5 +123,6 @@ func (c *criService) teardownPod(id string, path string, config *runtime.PodSand
 	return c.netPlugin.Remove(id,
 		path,
 		cni.WithLabels(labels),
-		cni.WithCapabilityPortMap(toCNIPortMappings(config.GetPortMappings())))
+		cni.WithCapabilityPortMap(toCNIPortMappings(config.GetPortMappings())),
+		cni.WithCapability("DNS", toCNIDNS(config.GetDnsConfig())))
```
Comment (madhanrm) [inline pkg/server/sandbox_stop.go]: Is this case sensitive "DNS"

File: pkg/server/sandbox_stop.go [hunk 2]
```diff
@@ -64,7 +64,7 @@ func (c *criService) StopPodSandbox(ctx context.Context, r *runtime.StopPodSandb
 			return nil, errors.Wrapf(err, "failed to stop sandbox container %q", id)
 		}
 	}
-
+
```
Comment (madhanrm) [inline pkg/server/sandbox_stop.go]: Nit: remove Blank spaces.

Comment (nagiesek) [reply pkg/server/sandbox_stop.go]: will do

File: pkg/server/sandbox_stop.go (hunk 1 above)
Comment (nagiesek) [reply pkg/server/sandbox_stop.go]: yes must be lowercase. for our stuff

File: pkg/server/helpers.go [hunk 3]
```diff
@@ -139,9 +139,9 @@ const (
 // generated is unique as long as sandbox metadata is unique.
 func makeSandboxName(s *runtime.PodSandboxMetadata) string {
 	return strings.Join([]string{
-		s.Name,      // 0
-		s.Namespace, // 1
-		s.Uid,       // 2
+		s.Name,                       // 0
```
Comment (madhanrm) [inline pkg/server/helpers.go]: Can you undo irrelevant changes?

File: pkg/server/helpers.go [hunk 4]
```diff
@@ -151,10 +151,10 @@ func makeSandboxName(s *runtime.PodSandboxMetadata) string {
 // unique.
 func makeContainerName(c *runtime.ContainerMetadata, s *runtime.PodSandboxMetadata) string {
 	return strings.Join([]string{
-		c.Name,      // 0
-		s.Name,      // 1: pod name
-		s.Namespace, // 2: pod namespace
-		s.Uid,       // 3: pod uid
+		c.Name,                       // 0
+		s.Name,                       // 1: pod name
```
Comment (madhanrm) [inline pkg/server/helpers.go]: Same as above

File: pkg/server/sandbox_stop.go (hunk 1 above)
Comment (madhanrm) [reply pkg/server/sandbox_stop.go]: So why is uppercase used ?

File: pkg/server/helpers.go (hunk 3 above)
Comment (nagiesek) [reply pkg/server/helpers.go]: This comes as a part of go fmt

File: pkg/server/sandbox_run.go [hunk 5]
```diff
@@ -111,7 +112,8 @@ func (c *criService) setupPod(id string, path string, config *runtime.PodSandbox
 	result, err := c.netPlugin.Setup(id,
 		path,
 		cni.WithLabels(labels),
-		cni.WithCapabilityPortMap(toCNIPortMappings(config.GetPortMappings())))
+		cni.WithCapabilityPortMap(toCNIPortMappings(config.GetPortMappings())),
+		cni.WithCapability("dns", toCNIDNS(config.GetDnsConfig())))
```
Comment (madhanrm) [inline pkg/server/sandbox_run.go]: Can you implement cni.WithCapabilityDns() in a follow up PR?

**Pull request merged**